src/content/ui-api/management-api/authentication/index.md

```diff
@@ -123,11 +123,87 @@ Let's go into details for the most relevant properties (also called "claims") of
+Each Giant Swarm installation provides a service account named `automation` in the `default` namespace that may be used for creating a self-contained `kubeconfig` file. **Note however**, this account comes with a quite powerful set of permissions, thus **we strongly recommend to create a specific service account for each application**, binding it to specific roles granting only the required permissions in the required namespaces.
+
+Regardless of service account you decide to use, and assuming you are currently logged in to the right installation, you can utilize the below instruction to create a config file.
+
+1. Export the Service Account name:
+
+```nohighlight
+SANAME=automation
+```
+
+2. Find Kubernetes Secret associated with the service account:
+
+```nohighlight
+SECRET=$(kubectl get sa $SANAME -o jsonpath='{.secrets[0].name}')
+```
+
+3. Find the service account's token:
+
+```nohighlight
+TOKEN=$(kubectl get secret $SECRET -o jsonpath='{.data.token}' | base64 --decode)
+```
+
+4. Find Kubernetes CA certificate:
+
+```nohighlight
+CACERT=$(kubectl get secret $SECRET -o jsonpath='{.data.ca\.crt}')
+```
+
+5. Grab installation's API endpoint:
+
+```nohighlight
+APIURL=$(kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}')
+```
+
+6. Grab installation's name:
+
+```nohighlight
+CODENAME=$(kubectl config view --minify -o jsonpath='{.clusters[0].name}')
+```
+
+7. Generate kubectl configuration:
+
+```nohighlight
+cat <<EOF > kubeconfig
+apiVersion: v1
+kind: Config
+clusters:
+  - name: $CODENAME
+    cluster:
+      certificate-authority-data: $CACERT
+      server: $APIURL
+users:
+  - name: $SANAME
+    user:
+      token: $TOKEN
+current-context: $CODENAME
+contexts:
+- context:
+    cluster: $CODENAME
+    user: $SANAME
+  name: $CODENAME
+EOF
+```
+
+8. Test newly created file:
+
+```nohighlight
+kubectl --kubeconfig kubeconfig cluster-info
+```
+
+### Remote access to the Workload Cluster {#remote-wc}
+
+Workload Clusters relies on the [TokenRequest API](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/#create-create-token-of-a-serviceaccount) for authentication, so instruction from the previous paragraph will not work against them. We strongly encourage you to use the `kubectl-gs` plugin and [client certificate login with a self-contained kubeconfig](https://docs.giantswarm.io/ui-api/kubectl-gs/login/#wc-client-cert-automation) instead.
+
+### Runners inside a cluster {#inside-runner}
+
+When you deploy any form of automation (e.g. runners) inside a cluster that is intended to perform work against this cluster, then we strongly recommend using [service accounts](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) directly. Please refer to your automation provider documentation for finding the right configuration steps.
```

Comment:
we could add links here to Gitlab or Jenkins as one of most popular tools, but not sure if that's not too much

Reply:
What is automation provider in this context? The CI/CD system? I think mentioning GitLab and Jenkins as examples is a good idea.

Reply:
Yup, I meant mostly the CI/CD system

Reply:
@ljakimczuk Cool I updated my suggestion below to "any form of automation (e.g. CI/CD runners)"

I think that's generic and then we don't need an example like GitLab. WDYT?

Reply:
Yeah, I agree, especially outside docs may get changed or moved around, so let's not risk breaking the links